### PR TITLE
42 group listing

### PR DESCRIPTION
**Pull request by falusi94**
Filter active groups by default and add possibility to list all groups.

### CODE REVIEW

File: test/controllers/groups_controller_test.rb [hunk 1]
```diff
@@ -5,14 +5,20 @@ class GroupsControllerTest < ActionController::TestCase
...
+    get :all, per: result_per_page
 
     assert_equal result_per_page, assigns(:groups).size
     assert_equal groups(:babhamozo).id, assigns(:groups).first.id
   end
 
+  test "show active groups list" do
+    get :index, per: 20
+
+    assert_equal 0, assigns(:groups).size
```
Comment (emanreyalp) [inline test/controllers/groups_controller_test.rb]: 20-at kerunk de 0-t kapunk? ha bovul a fixtures akkor valszeg mar nem 0-t kapunk.

File: app/controllers/groups_controller.rb [hunk 2]
```diff
@@ -3,7 +3,15 @@ class GroupsController < ApplicationController
   before_action :require_leader, only: [:edit, :update]
 
   def index
+    active_groups = Group.order(:name).select{ |g| !g.inactive? }
+    active_groups = GroupDecorator.decorate_collection(active_groups)
+    @groups = Kaminari.paginate_array(active_groups)
+      .page(params[:page]).per(params[:per])
+  end
+
+  def all
```
Comment (emanreyalp) [inline app/controllers/groups_controller.rb]: Miert szedjuk kulon?
Szerintem szebb az ha siman csak aktivitas alapjan sorrendezzuk. Esetleg maskent jelenitsuk meg az aktivakat az inaktivtol. 
De ha mindenkepp kulon akarjuk, szerintem akkor is maradhatna az indexben. Es a view-ba raknek egy kapcsolot amivel az osszeset vagy csak az aktivakat mutatjuk.

Comment (falusi94) [reply app/controllers/groups_controller.rb]: With separation by url you can easily preserve the previous state of the page, if users want to link it or use browsers back button.

File: test/controllers/groups_controller_test.rb (hunk 1 above)
Comment (falusi94) [reply test/controllers/groups_controller_test.rb]: Inactive? checks the last two semesters' evaulation and now none of the test groups have any. So in my opinion, the test works fine :D

Comment (emanreyalp) [reply test/controllers/groups_controller_test.rb]: But if sb add evaulation 'cause wanna test is, then the test will fail.

File: app/controllers/groups_controller.rb (hunk 2 above)
Comment (emanreyalp) [reply app/controllers/groups_controller.rb]: In that case, we use query string in url sth like '/?show=all' or '?show=active?'

Comment (salierri) [reply app/controllers/groups_controller.rb]: I don't entirely agree with this. Query strings are meant for passing variables that can take any values (ex. search query, or page count). This is not the case here, so even if one route handles both options the route could be something like `/show/all` and `/show/active`.

And yea, I know I'm a bit late with this :smile:

**Pull request merged**